lib/Controller/MyOCSController.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace OCA\Tables\Controller;
+
+use Exception;
+use OCA\Tables\AppInfo\Application;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\NotFoundError;
+use OCA\Tables\Errors\PermissionError;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\DataResponse;
+use OCP\AppFramework\OCSController;
+use OCP\IRequest;
+use Psr\Log\LoggerInterface;
+
+class MyOCSController extends OCSController {
+
+	protected LoggerInterface $logger;
+	protected string $userId;
+
+	public function __construct(
+		IRequest $request,
+		LoggerInterface $logger,
+		string $userId
+	) {
+		parent::__construct(Application::APP_ID, $request);
+		$this->logger = $logger;
+		$this->userId = $userId;
+	}
+
+	/**
+	 * @param Exception|InternalError $e
+	 * @return DataResponse<Http::STATUS_INTERNAL_SERVER_ERROR, array{message: string}, array{}>
+	 */
+	protected function handleError(InternalError|Exception $e): DataResponse {
+		$this->logger->warning('An internal error or exception occurred: ' . $e->getMessage());
+		return new DataResponse(['message' => $e->getMessage()], Http::STATUS_INTERNAL_SERVER_ERROR);
```

Comment:
We should avoid passing the exception message to the user as it may contain security related information. 

Best would be to use https://github.com/nextcloud/server/blob/d55a7c619d3733863387cc8c6269050b72cbff0c/lib/public/HintException.php#L76 for custom exception messages that you want to expose.